libqtile/layout/columns.py

```diff
@@ -205,6 +205,24 @@ def remove(self, client):
             self.remove_column(c)
         return self.columns[self.current].cw
 
+    def swap(self, client1, client2):
+        column1 = None
+        column2 = None
+        for c in self.columns:
+            if client1 in c:
+                column1 = c
+            if client2 in c:
+                column2 = c
+        if not (column1 and column2):
+            return
+        if column1 is column2:
+            column1.swap(client1, client2)
+        else:
+            column1.add(client2)
+            column2.add(client1)
+            column1.remove(client1)
+            column2.remove(client2)
```

Comment:
In this case, shouldn't each client be added into the position from where the other was removed? It seems to me that the order will be messed up. It would also reset the relative sizes between clients in each column. The same problem would happen with `Stack`. I believe this should be replaced by (I have not tested it , though):
```python
i1 = column1.index(client1) - column1.current_index
i2 = column2.index(client2) - column2.current_index
h1 = column1.heights[client1]
h2 = column2.heights[client2]
temp = column1.insert_position
column1.insert_position = i1
column1.add(client2, h2)
column1.insert_position = temp
temp = column2.insert_position
column2.insert_position = i2
column2.add(client1, h1)
column2.insert_position = temp
column1.remove(client1)
column2.remove(client2)
```